streaming/rdm_atomic.c

```diff
@@ -190,44 +227,98 @@ static int execute_compare_atomic_op(enum fi_op op)
 	return ret;
 }
 
+static void report_perf()
+{
+	if (opts.machr)
+		show_perf_mr(opts.transfer_size, opts.iterations, &start, &end, 1, opts.argc,
+			opts.argv);
+	else
+		show_perf(test_name, opts.transfer_size, opts.iterations, &start, &end, 1);
+}
+
 static int run_op(void)
 {
-	int ret, i;
+	int ret, i, len;
 
 	count = (size_t *) malloc(sizeof(size_t));
 	ft_sync();
 
-
-	ft_start();
 	switch (op_type) {
 	case FI_MIN:
 	case FI_MAX:
+	case FI_SUM:
+	case FI_PROD:
+	case FI_LOR:
+	case FI_LAND:
+	case FI_BOR:
+	case FI_BAND:
+	case FI_LXOR:
+	case FI_BXOR:
 	case FI_ATOMIC_WRITE:
-		ret = is_valid_base_atomic_op(op_type);
-		if (ret > 0) {
-			for (i = 0; i < opts.iterations; i++) {
-				ret = execute_base_atomic_op(op_type);
-				if (ret)
-					break;
+		for (datatype = 0; datatype <= FI_LONG_DOUBLE_COMPLEX; datatype++) {
+			ft_start();
+			ret = is_valid_base_atomic_op(op_type);
+			if (ret > 0) {
+				len = snprintf(test_name, sizeof(test_name), "%s_",
+					fi_tostr(&datatype, FI_TYPE_ATOMIC_TYPE));
+				snprintf(test_name + len, sizeof(test_name), "%s_base_lat",
+					fi_tostr(&op_type, FI_TYPE_ATOMIC_OP));
+				opts.transfer_size = datatype_to_size(datatype);
+				init_test(&opts, test_name, sizeof(test_name));
+				for (i = 0; i < opts.iterations; i++) {
+					ret = execute_base_atomic_op(op_type);
+					if (ret)
+						break;
+				}
+				ft_stop();
```

Comment:
The start and stop calls are not aligned to the same level, making it look unbalanced.  The start call should not occur until immediately before the inner for loop.

Also consider changing the if (ret > 0) to if (ret <=0) continue, and out-dent the code.